internal/controllers/clusterextension_controller.go

```diff
@@ -273,9 +273,9 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 	// Generate a BundleDeployment from the ClusterExtension to Unpack.
 	// Note: The BundleDeployment here is not a k8s API, its a simple Go struct which
 	// necessary embedded values.
-	bd := r.generateBundleDeploymentForUnpack(ctx, bundle.Image, ext)
+	bd := r.generateBundleDeploymentForUnpack(bundle.Image, ext)
 	l.V(1).Info("unpacking resolved bundle")
-	unpackResult, err := r.Unpacker.Unpack(ctx, bd)
+	unpackResult, err := r.Unpacker.Unpack(ctx, bd, r.CaCertPool)
```

Comment:
Is this the only place that `r.CaCertPool` is used in the reconciler? If so, WDYT about moving this to a field on the `Unpack` implementation's struct?

Reply:
(I had problems with mocks, so the more I can stay away from them...)